modules/extensions-support/src/main/java/org/mule/module/extension/internal/util/NameUtils.java

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.util;
+
+import org.mule.extension.annotations.Alias;
+import org.mule.extension.introspection.DataType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+
+public class NameUtils
+{
+
+    private static final List<Inflection> plural = new ArrayList<>();
+    private static final List<Inflection> singular = new ArrayList<>();
+    private static final List<String> uncountable = new ArrayList<>();
+
+    //TODO: Refactor this. This was fine on devkit but we can't keep all these static info in runtime
+    static
```

Comment:
There is no somethign alredy available for this? (seems like spring does something similar)